libs/utils/wa_results_collector.py

```diff
@@ -298,8 +303,11 @@ def _read_wa_dir(self, wa_dir):
             extra_df.loc[:, 'id'] = job_id
             extra_df.loc[:, 'tag'] = tag
             extra_df.loc[:, 'test'] = test
+            # Collect all thse DFs to merge them in one go at the end.
+            extra_dfs.append(extra_df)
 
-            df = df.append(extra_df)
+        # Append all extrace DFs to the results WA's results DF
+        df = df.append(extra_dfs)
```

Comment:
I would get an exception here for a wltest run of 2 kernels and only Geekbench. This fixed it:
```
if extra_dfs:
    df = df.append(extra_dfs)
```

Reply:
Maybe, just to be more clear, if `len(extra_dfs)` ?

Reply:
I don't mind either, but the Holy Book says:
"""
For sequences, (strings, lists, tuples), use the fact that empty sequences are false.

```
Yes: if not seq:
     if seq:

No: if len(seq):
    if not len(seq):
```
"""


(https://www.python.org/dev/peps/pep-0008/)

Reply:
But actually... according to this:
   https://docs.python.org/2/library/stdtypes.html#truth-value-testing
what you propose is a legitimate python "nonsense"...